allennlp/predictors/bidaf.py

```diff
@@ -38,3 +44,101 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
+        answer_as_add_sub_expressions, answer_as_counts. We need to provide labels for all of them.
+        """
+        new_instance = deepcopy(instance)
+        # For BiDAF
+        if 'best_span' in outputs:
+            span_start_label = outputs['best_span'][0]
+            span_end_label = outputs['best_span'][1]
+            new_instance.add_field('span_start',
+                                   IndexField(int(span_start_label), new_instance['passage'])) # type: ignore
+            new_instance.add_field('span_end',
+                                   IndexField(int(span_end_label), new_instance['passage'])) # type: ignore
+
+        # For NAQANet model
+        elif 'answer' in outputs:
+            answer_type = outputs['answer']['answer_type']
+
+            # When the problem is a counting problem
+            if answer_type == 'count':
+                field = ListField([LabelField(int(outputs['answer']['count']), skip_indexing=True)])
+                new_instance.add_field('answer_as_counts', field)
+
+            # When the answer is in the passage
+            elif answer_type == 'passage_span':
+                span = outputs['answer']['spans'][0]
+
+                # Convert character span indices into word span indices
+                word_span_start = None
+                word_span_end = None
+                offsets = new_instance['metadata'].metadata['passage_token_offsets'] # type: ignore
+                for idx, offset in enumerate(offsets):
+                    if offset[0] == span[0]:
+                        word_span_start = idx
+                    if offset[1] == span[1]:
+                        word_span_end = idx
+
+                passage_field: SequenceField = new_instance['passage']  # type: ignore
+                field = ListField([SpanField(word_span_start,
+                                             word_span_end,
+                                             passage_field)]) # type: ignore
+                new_instance.add_field('answer_as_passage_spans', field)
+
+            # When the answer is an arithmetic calculation
+            elif answer_type == 'arithmetic':
+                # The different numbers in the passage that the model encounters
+                sequence_labels = outputs['answer']['numbers']
+
+                numbers_as_tokens = \
+                [Token(str(number['value'])) for number in outputs['answer']['numbers']]
+                # hack copied from https://github.com/
+                # allenai/allennlp/blob/master/allennlp/
+                # data/dataset_readers/reading_comprehension/drop.py#L232
+                numbers_as_tokens.append(Token('0'))
+                numbers_in_passage_field = TextField(numbers_as_tokens,
+                                                     self._dataset_reader._token_indexers) # type: ignore
+
+                # Based on ``find_valid_add_sub_expressions``, it seems
```

Comment:
Remove "it seems like", and just say, "The class id for negative numbers is 2", or something like that.